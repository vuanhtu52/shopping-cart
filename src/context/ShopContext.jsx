import { createContext } from "react";

const ShopContext = createContext({
    products: [],
    cartItems: [],
    addToCart: () => {},
    addMultipleToCart: () => {},
    removeOneFromCart: () => {},
    getTotalCartItems: () => {},
});

export default ShopContext;