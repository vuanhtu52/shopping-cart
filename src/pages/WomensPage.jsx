const WomensPage = () => {
    return <div>Women's Clothing</div>;
};

export default WomensPage;
