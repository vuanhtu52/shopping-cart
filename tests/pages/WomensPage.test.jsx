import { render, screen } from "@testing-library/react";
import ShopContext from "../../src/context/ShopContext";
import { MemoryRouter } from "react-router-dom";
import WomensPage from "../../src/pages/WomensPage";
import { describe, expect } from "vitest";

const renderComponent = products => {
    return render(
        <ShopContext.Provider value={{products}}>
            <MemoryRouter>
                <WomensPage />
            </MemoryRouter>
        </ShopContext.Provider>
    );
};

describe("WomensPage component", () => {
    it("Renders heading correctly", () => {
        renderComponent([]);
        expect(screen.getByText("Women's Clothing")).toBeInTheDocument();
    });

    it("Renders only women's clothing", () => {
        const products = [
            { id: 1, title: "Dress", category: "women's clothing" },
            { id: 2, title: "Jeans", category: "men's clothing" }, // Should be ignored
            { id: 3, title: "Blouse", category: "women's clothing" }
        ];

        renderComponent(products);

        expect(screen.getByText("Dress")).toBeInTheDocument();
        expect(screen.queryByText("Jeans")).not.toBeInTheDocument();
        expect(screen.getByText("Blouse")).toBeInTheDocument();
    });

    it("Renders no products when none are women's clothing", () => {
        const products = [
            { id: 1, title: "Jeans", category: "men's clothing" },
            { id: 2, title: "T-Shirt", category: "electronics" }
        ];

        renderComponent(products);

        expect(screen.queryByText("Jeans")).not.toBeInTheDocument();
        expect(screen.queryByText("T-Shirt")).not.toBeInTheDocument();
    });

    it("Snapshot testing", () => {
        const products = [
            { id: 1, title: "Jeans", category: "men's clothing" },
            { id: 2, title: "T-Shirt", category: "electronics" }
        ];

        const {container} = renderComponent(products);
        expect(container).toMatchSnapshot();
    });
});