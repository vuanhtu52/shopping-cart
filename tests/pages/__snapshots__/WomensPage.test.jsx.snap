// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WomensPage component > Snapshot testing 1`] = `
<div>
  <div>
    <h2
      class="text-center text-xl font-bold p-4"
    >
      Women's Clothing
    </h2>
    <div
      class="grid gap-8 justify-items-center grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 p-4"
    />
  </div>
</div>
`;
